manager/integration/tests/test_basic.py

```diff
@@ -3324,28 +3326,73 @@ def test_allow_volume_creation_with_degraded_availability_error(
     check_volume_data(volume, data)
 
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_multiple_volumes_creation_with_degraded_availability():
+def test_multiple_volumes_creation_with_degraded_availability(
+    set_random_backupstore, client, core_api, apps_api, storage_class, statefulset):  # NOQA
     """
     Goal:
     We want to verify that multiple volumes with degraded availability
     can be created, attached, detached, and deleted at the nearly the
     same time.
 
     Steps:
-    1. create StorageClass longhorn-extra with numberOfReplicas=5
-       Set allow-volume-creation-with-degraded-availability to True
-    2. Deploy this StatefulSet:
+    1. create new StorageClass with numberOfReplicas=5.
+    2. Set allow-volume-creation-with-degraded-availability to True.
+    3. Deploy this StatefulSet:
        https://github.com/longhorn/longhorn/issues/2073#issuecomment-742948726
-    3. In a 1-min retry loop, Verify that all 10 volumes are healthy
-    4. Delete the StatefulSet
-    5. In a 1-min retry loop, Verify that all 10 volumes are detached
-    6. Find and delete the PVC of the 10 volumes.
-    7. In a 1-min retry loop, Verify that all 10 volumes are deleted
-    8. Make sure to delete all extra storage classes in
-       common.cleanup_client()
-    """
-    pass
+    4. In a 1-min retry loop, Verify that all 10 volumes are healthy.
+    5. Delete the StatefulSet.
+    6. In a 1-min retry loop, Verify that all 10 volumes are detached.
+    7. Find and delete the PVC of the 10 volumes.
+    8. In a 1-min retry loop, Verify that all 10 volumes are deleted.
+    9. Make sure to delete all extra storage classes in
+       common.cleanup_client().
+    """
+    storage_class['parameters']['numberOfReplicas'] = "5"
+    create_storage_class(storage_class)
+
+    setting = client.by_id_setting(common.SETTING_DEGRADED_AVAILABILITY)
+    client.update(setting, value="true")
+
+    sts_spec = statefulset['spec']
+    sts_spec['podManagementPolicy'] = "Parallel"
+    sts_spec['replicas'] = 10
+    sts_spec['volumeClaimTemplates'][0]['spec']['storageClassName'] = \
+        storage_class['metadata']['name']
+    common.create_statefulset(statefulset)
+
+    retry_count = int(60 / RETRY_INTERVAL)
+    for _ in range(common.DEFAULT_STATEFULSET_TIMEOUT):
+        try:
+            pod_info = common.get_statefulset_pod_info(core_api, statefulset)
+            break
+        except ApiException as e:
+            assert e.status == 404
+            common.DEFAULT_STATEFULSET_INTERVAL
+    assert pod_info is not None
```

Comment:
Why is `create_and_wait_statefulset` not used?

And you can add check `assert len(pod_list) == 10`

Reply:
> Why is `create_and_wait_statefulset` not used?

`create_and_wait_statefulset` loop through 15 mins to wait for replicas to be ready, this looks like it contradicts with the testing step `In a 1-min retry loop, Verify that all 10 volumes are healthy`.

Reply:
Why does the function need to wait for 15min? It's too long for a single test... You can directly modify the function if possible.

Reply:
We can modify the function and it will still wait for replicas to be ready and won't volume already be healthy?

Reply:
@PhanLe1010 since you are the original writer of this test skeleton, can we have your option on this?

Reply:
Updated to use `create_and_wait_statefulset` and also update the `DEFAULT_STATEFULSET_INTERVAL` to 1, similar to the timeout for pod and deployment.